core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -857,37 +664,75 @@ public static boolean isValid(String keyStr) {
    * @return corresponding property
    */
   public static PropertyKey fromString(String keyStr) {
-    PropertyKey key = KEYS_MAP.get(keyStr);
-    if (key == null) {
-      throw new IllegalArgumentException("Invalid property key " + keyStr);
+    PropertyKey key = ConstantPropertyKey.fromString(keyStr);
+    if (key != null) {
+      return key;
     }
-    return key;
+    key = ParameterizedPropertyKey.fromString(keyStr);
+    if (key != null) {
+      return key;
+    }
+    throw new IllegalArgumentException(
+        ExceptionMessage.INVALID_CONFIGURATION_KEY.getMessage(keyStr));
   }
 
   /**
-   * @return a Map from property key name to the key
+   * @return all pre-defined property keys
    */
-  private static Map<String, PropertyKey> initKeysMap() {
-    Map<String, PropertyKey> map = new HashMap<>();
-    for (PropertyKey key : PropertyKey.values()) {
-      map.put(key.toString(), key);
-    }
-    return map;
+  public static Collection<ConstantPropertyKey> getDefaultKeys() {
```

Comment:
changed to be `Collection<? extends PropertyKey>`. Otherwise it requires to enumerate the `Colllection<ConstantPropertyKey>` and convert each item to `PropertyKey` individually.